llvm_util/utils.cpp

```diff
@@ -165,6 +170,34 @@ Value* get_operand(llvm::Value *v) {
     return ret;
   }
 
+  if (auto gv = dyn_cast<llvm::GlobalVariable>(v)) {
+    auto itr = globalvars.find(gv);
+    if (itr != globalvars.end())
+      return itr->second;
+
+    if (gv->getValueType()->isArrayTy()) {
+      // TODO: Array type of a global variable is not supported.
+      return nullptr;
+    } else if (auto st = dyn_cast<llvm::StructType>(gv->getValueType())) {
+      if (st->isOpaque())
+        // TODO: Global variable of opaque type is not supported.
+        return nullptr;
+    }
+    Value *initval = nullptr;
+    if (gv->hasInitializer()) {
+      initval = get_operand(gv->getInitializer());
```

Comment:
pls don't the initializer if it isn't const

Reply:
updated